plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/CloudToolsInfo.java

```diff
@@ -16,20 +16,38 @@
 
 package com.google.cloud.tools.eclipse.util;
 
-import org.osgi.framework.FrameworkUtil;
+import org.eclipse.core.runtime.IBundleGroup;
+import org.eclipse.core.runtime.IBundleGroupProvider;
+import org.eclipse.core.runtime.Platform;
 
 /**
  * Provides generic information about the plug-in, such as a name to be used for usage
  * reporting and the current version, etc.
  */
 public class CloudToolsInfo {
 
+  /**
+   * Our main feature identifier, used for branding.
+   */
+  private static final String CLOUD_TOOLS_FOR_ECLIPSE_FEATURE_ID =
+      "com.google.cloud.tools.eclipse.suite.e45.feature";
+
   // Don't change the value; this name is used as an originating "application" of usage metrics.
   public static String METRICS_NAME = "gcloud-eclipse-tools";
 
   public static String USER_AGENT = METRICS_NAME + "/" + getToolsVersion();
 
+  /** Return the version of associated feature. May be slow. */
```

Comment:
I think I'm being overly cautious.  It's certainly not a network fetch.  I think caching it is fine: realistically if someone updates the feature, they'll restart.